client/src/js/settings/components/Jobs/Resources.js

```diff
@@ -1,59 +1,193 @@
 import React from "react";
 import { connect } from "react-redux";
-import { Row, Col, Panel } from "react-bootstrap";
+import { Row, Col, Panel, Alert } from "react-bootstrap";
+import { toNumber } from "lodash-es";
 
 import { updateSetting } from "../../actions";
-import { InputError } from "../../../base";
-
-
-const Resources = (props) => (
-    <Row>
-        <Col md={12}>
-            <h5><strong>Resource Limits</strong></h5>
-        </Col>
-        <Col xs={12} md={6} mdPush={6}>
-            <Panel>
-                Set limits on the computing resources Virtool can use on the host server.
-            </Panel>
-        </Col>
-        <Col xs={12} md={6} mdPull={6}>
-            <Panel>
-                <InputError
-                    label="CPU Limit"
-                    type="number"
-                    onSave={props.onUpdateProc}
-                    initialValue={props.proc}
-                    withButton
-                />
-                <InputError
-                    label="Memory Limit (GB)"
-                    type="number"
-                    onSave={props.onUpdateMem}
-                    initialValue={props.mem}
-                    withButton
-                />
-            </Panel>
-        </Col>
-    </Row>
-);
+import { getResources } from "../../../jobs/actions";
+import { InputError, LoadingPlaceholder, Icon} from "../../../base";
+
+class Resources extends React.Component {
+
+    constructor (props) {
+        super(props);
+
+        this.state = {
+            errorProc: false,
+            errorMem: false,
+            procUpperLimit: this.props.proc,
```

Comment:
Is it necessary to move `procUpperLimit` and `memUpperLimit` into state?

I think you are doing it to deal with lag between the component loading and the `getResources` request returning...